Hpc - Guides creation VM encryption on vCenter with OKMS (#6485)
ovh/docs: pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/guide.fr-fr.md

---

> title:'Activation du chiffrement de VM avec le KMS OVHcloud'

Reviewer: ```suggestion
title: "Activer le chiffrement de VM avec le KMS OVHcloud"
```

> excerpt:'Découvrez comment protégez votre confidentialité et activer le chiffrement de vos machines virtuelles Hosted Private Cloud avec la solution KMS OVHcloud'

Reviewer: ```suggestion
excerpt: "Découvrez comment protéger votre confidentialité et activer le chiffrement de vos machines virtuelles Hosted Private Cloud avec la solution KMS OVHcloud"
```

> updated: 2024-07-18

Reviewer: ```suggestion
updated: 2024-07-19
```

---
<style>
details>summary {
    color:rgb(33, 153, 232) !important;
    cursor: pointer;
}
details>summary::before {
    content:'\25B6';
    padding-right:1ch;
}
details[open]>summary::before {
    content:'\25BC';
}
</style>

> [!primary]
>

> > Cette fonctionnalité est disponible en bêta. Ce guide peut être incomplet et sera mis à jour dans les versions futures. N’hésitez pas à nous faire part de vos feedbacks sur le canal [Discord](<https://discord.gg/ovhcloud>) dédié.

Reviewer: ```suggestion
> Cette fonctionnalité est disponible en version bêta. Ce guide peut donc être incomplet et sera mis à jour pendant cette période. N’hésitez pas à nous faire part de votre expérience sur le canal [Discord](<https://discord.gg/ovhcloud>) dédié.
```

>

## Objectif

> **Commander, activer et configurer un KMS OVHcloud (Okms) au sein de vSphere pour activer la politique de chiffrement de vos machines virtuelles Hosted Private Cloud VMware on OVHcloud.**

Reviewer: ```suggestion
**Découvrez comment commander, activer et configurer un KMS OVHcloud (OKMS) au sein de vSphere pour gérer la politique de chiffrement de vos machines virtuelles Hosted Private Cloud VMware on OVHcloud.**
```

## Prérequis

> - Disposer d'un [compte client OVHcloud](/pages/account_and_service_management/account_information/ovhcloud-account-creation).

Reviewer: ```suggestion
- Être connecté à votre [espace client OVHcloud](/links/manager).
```

> - Avoir souscrit une offre [VMware on OVHcloud](https://www.ovhcloud.com/fr/enterprise/products/hosted-private-cloud/){.external}.

Reviewer: ```suggestion
- Disposer d'une offre [VMware on OVHcloud](https://www.ovhcloud.com/fr/enterprise/products/hosted-private-cloud/){.external}.
```

- Avoir accès à l’interface de gestion vSphere de votre PCC (Hosted Private Cloud VMware on OVHcloud).
- Disposer des droits nécessaires pour manipuler vos ressources PCC et le KMS OVHcloud avec ou sans IAM (IAM with KMS: iam_resource_type_okms dans la policy IAM + ajouter l'id dans vos ressources de la policy IAM).
- Avoir lu les guides : 

>   - [Que KMS au sein de Hosted Private Cloud VMware on OVHcloud](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/kms_vmware_overall).

Reviewer: ```suggestion
  - [Intégration d'un KMS pour VMware on OVHcloud](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/kms_vmware_overall).
```

>   - [Premier pas (KMS + IAM + vSphere)](/pages/manage_and_operate/kms/quick-start).

Reviewer: ```suggestion
  - [Premiers pas avec OVHcloud Key Management Service (KMS + IAM + vSphere)](/pages/manage_and_operate/kms/quick-start).
```

> - Une clef de chiffrement RSA ou ECDSA (sans CSR) et/ou un CSR (avec CSR).

Reviewer: ```suggestion
- Avoir une clef de chiffrement RSA ou ECDSA (sans CSR) et/ou un CSR (avec CSR).
```

> - Le certificat public tls du serveur Okms afin de la faire signer par le Okms.

Reviewer: ```suggestion
- Disposer du certificat public TLS du serveur OKMS afin de le faire signer par l'OKMS.
```

## En pratique

> /// details | Introduction, listing url et les appels api okms disponibles.

Reviewer: ```suggestion
/// details | Introduction, liste des URLs et des appels API OKMS disponibles.
```

> Pour plus d'information sur les choix qui s'offre à vous avec KMS et Hosted Private Cloud VMware on OVHcloud, lisez le guide [Activer KMS au sein de Hosted Private Cloud VMware on OVHcloud](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/kms_vmware_overall).

Reviewer: ```suggestion
Pour plus d'informations sur les choix qui s'offrent à vous avec KMS et Hosted Private Cloud VMware on OVHcloud, consultez le guide « [Intégration d'un KMS pour VMware on OVHcloud](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/kms_vmware_overall) ».
```

> Nous allons dans ce guide configurer le chiffrement de machines virtuelles avec le KMS OVHcloud OKMS au sein d'un environnement Hosted Private Cloud VMware on OVHcloud avec le KMS OVHcloud (OKMS).

Reviewer: ```suggestion
Nous allons, dans ce guide, configurer le chiffrement de machines virtuelles avec le KMS OVHcloud (OKMS) au sein d'un environnement Hosted Private Cloud VMware on OVHcloud.
```

> OVHcloud KMS (Okms) est le service centralisé de chiffrement entièrement managé pour sécuriser vos données dans vos applications et vos services OVHcloud.

Reviewer: ```suggestion
OVHcloud KMS (OKMS) est un service centralisé de chiffrement entièrement managé pour sécuriser vos données dans vos applications et vos services OVHcloud.
```

> ### Urls Okms

Reviewer: ```suggestion
### URLs OKMS
```

> | **Type**    | **Url**                          | **Région**          | **Okms Région enum** | **IP**         |

Reviewer: ```suggestion
| **Type**    | **URL**                          | **Région**          | **OKMS Région enum** | **IP**         |
```

|-------------|----------------------------------|---------------------|----------------------|----------------|
| **KMIP**    | eu-west-rbx.okms.ovh.net         | France - Roubaix    | EU_WEST_RBX          | 91.134.128.102 |
| **KMIP**    | eu-west-rbx.okms.ovh.net         | France - Roubaix    | EU_WEST_RBX          | 91.134.128.102 |
| **REST**    | eu-west-rbx.okms.ovh.net         | France - Roubaix    | EU_WEST_RBX          | 91.134.128.102 |
| **Swagger** | swagger-eu-west-rbx.okms.ovh.net | France - Roubaix    | EU_WEST_RBX          | 91.134.128.102 |
| **KMIP**    | eu-west-sbg.okms.ovh.net         | France - Strasbourg | EU_WEST_SBG          | 137.74.127.152 |
| **KMIP**    | eu-west-sbg.okms.ovh.net         | France - Strasbourg | EU_WEST_SBG          | 137.74.127.152 |
| **REST**    | eu-west-sbg.okms.ovh.net         | France - Strasbourg | EU_WEST_SBG          | 137.74.127.152 |
| **Swagger** | swagger-eu-west-sbg.okms.ovh.net | France - Strasbourg | EU_WEST_SBG          | 137.74.127.152 |

> ## Listing des appels api KMS HPC VMware on OVHcloud

Reviewer: ```suggestion
### Liste des appels API KMS HPC VMware on OVHcloud
```

> | **Methode**           | **Api** | **Path**                                                                | **Commentaires**                                                |

Reviewer: ```suggestion
| **Methode**           | **API** | **Path**                                                                | **Commentaires**                                                |
```

> |-----------------------|---------|-------------------------------------------------------------------------|-----------------------------------------------------------------|

Reviewer: I think the dashboard should be reoganized.

| **Authentification:** |         |                                                                         |                                                                 |
| **GET**               | v1      | /dedicatedCloud/{serviceName}/vmEncryption/kms                          | - List virtual machine encryption KMS servers.                  |
| **POST**              | v1      | /dedicatedCloud/{serviceName}/vmEncryption/kms                          | - Create virtual machine encryption KMS server.                 |
| **GET**               | v1      | /dedicatedCloud/{serviceName}/vmEncryption/kms/{kmsId}                  | - Get virtual machine encryption KMS server.                    |
| **DEL**               | v1      | /dedicatedCloud/{serviceName}/vmEncryption/kms/{kmsId}                  | - Remove virtual machine encryption KMS server.                 |
| **POST**              | v1      | /dedicatedCloud/{serviceName}/vmEncryption/kms/{kmsId}/changeProperties | - Update virtual machine encryption KMS server.                 |

///

> ## Étape 1 - Commande d'un KMS OVHcloud (Okms) au sein de HPC VMware on OVHcloud (obligatoire)

Reviewer: ```suggestion
### Étape 1 - Commander un KMS OVHcloud (OKMS) au sein de HPC VMware on OVHcloud (obligatoire)
```

/// details | Comment commander un KMS OVHcloud pour HPC VMware on OVHcloud?

> ![Manager kms Menu Order Animated Gif](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_order.gif){.thumbnail}.

Reviewer: ```suggestion
![Manager KMS Menu Order Animated Gif](images/okms_order.gif){.thumbnail}.
```

> ### Via le control panel OVHcloud

Reviewer: ```suggestion
#### Via l'espace client OVHcloud
```

> Pour commander un fournisseur de clés KMS OVHcloud depuis le control panel Hosted Private Cloud on OVHcloud.

Reviewer: ```suggestion
Pour commander un fournisseur de clés KMS OVHcloud, connectez-vous à votre [espace client OVHcloud](/links/manager), puis rendez-vous dans la partie `Hosted Private Cloud`.{.action}. Dans la colonne de gauche, cliquez sur `Identité, Sécurité & Opération`.{.action}, puis sur `Key Management Service`.{.action}.
```

> 

Reviewer: ```suggestion
```

> Vous devez vous connecter au: [Control Panel OVHcloud](/links/manager).

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> Vous êtes invité à vous rendre dans le service KMS: 

Reviewer: ```suggestion
```

> - `Hosted Private Cloud > Identité, Sécurité & Opérations > Key Management Service`.{.action}

Reviewer: ```suggestion
```

> Si vous n'avez pas de serveur KMS, cliquez sur: `Commander un KMS`{.action}.

Reviewer: ```suggestion
Pour commander un nouveau serveur KMS, cliquez sur le bouton `Commander un KMS`{.action}, puis `Sélectionnez une région`{.action} parmi les deux disponibles actuellement :
```

> 

Reviewer: ```suggestion
```

> Vous êtes dans "Commander un KMS", choisissez la région.

Reviewer: ```suggestion
```

> Les clés de chiffrement et certificats d’accès de ce KMS seront stockées dans la région indiquée. Ils pourront être utilisés dans tous les produits OVHcloud sans distinction de région.

Reviewer: ```suggestion

- `Europe - France Roubaix`{.action}.
- `Europe - France Strasbourg`{.action}.
```

> Vous disposez des régions suivantes à ce jour disponible : 

Reviewer: ```suggestion
Les clés de chiffrement et certificats d’accès de ce KMS seront stockées dans la région indiquée. Ils pourront être utilisés dans tous les produits OVHcloud sans distinction de région.
```

> 

Reviewer: ```suggestion
```

> - `Europe - France Roubaix`{.action}.

Reviewer: ```suggestion
```

> - `Europe - France Strasbourg`{.action}.

Reviewer: ```suggestion
```

> Une fois votre choix fait, cliquez sur: `Commander`{.action}.

Reviewer: ```suggestion
Une fois votre choix fait, cliquez sur le bouton `Commander`{.action}.
```

> Si vous n'avez pas pu completer la commande, lancez [ce lien](https://www.ovh.com/fr/order/express/#/express/review?products=~(~(productId~'okms~planCode~'okms~duration~'P1M~pricingMode~'default~configuration~(~(label~'region~value~'EU_WEST_RBX))))).

Reviewer: ```suggestion
> [!primary]
> Si vous n'avez pas réussi à completer la commande, lancez [ce lien](https://www.ovh.com/fr/order/express/#/express/review?products=~(~(productId~'okms~planCode~'okms~duration~'P1M~pricingMode~'default~configuration~(~(label~'region~value~'EU_WEST_RBX))))).
>
```

> Puis, pour finir sur: `Terminer`{.action}.

Reviewer: ```suggestion
Enfin, cliquez sur `Terminer`{.action} pour finaliser la commande.
```

> ![Manager kms Menu Okms Order Finish](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_order_finish-optimized.png){.thumbnail}

Reviewer: ```suggestion
![Manager kms Menu OKMS Order Finish](images/okms_order_finish-optimized.png){.thumbnail}
```

> Une fois votre commande validée, vous retrouverez avec votre KMS, le **Nom**, l'**Id** et la **Région**.

Reviewer: ```suggestion
Une fois votre commande validée, vous retrouverez avec votre KMS, le **Nom**, l'**ID** et la **Région**.
```

> ![Manager kms Menu Kms All 01](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_okms_all-optimized.png){.thumbnail}

Reviewer: ```suggestion
![Manager KMS Menu KMS All 01](images/manager_okms_all-optimized.png){.thumbnail}
```

> Vous pouvez récupérer (copier-coller) votre "**okmsId**", en cliquant sur le bouton de droite qui le permet (copier) : ![Manager kms Menu](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_okms_copy_id-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
Copiez votre "**ID**" OKMS.

![Manager KMS Menu](images/manager_okms_copy_id-optim-resize.png){.thumbnail}
```

> L'id resemble à un **uuid** classique et vous sera utile par la suite pour la demande de nouvelles informations d'identification d'accès avec et/ou sans CSR ainsi que pour lister vos ressources Okms.

Reviewer: ```suggestion
L'ID resemble à un **UUID** classique. Il sera utile par la suite pour : 

 - la demande de nouvelles informations d'identification d'accès avec ou sans CSR ;
 - lister vos ressources OKMS.
```

> [!api]
> 
> @api {v2} /okms POST /okms/resource/{okmsId}/credential
>

> **Paramètres:**
> 

> > - okmsId: L'ID de votre KMS OVHcloud (Okms). 

Reviewer: ```suggestion
> - `okmsId`: L'ID de votre KMS OVHcloud (OKMS). 
```

>

Nous verrons dans l'étape 3 comment effectuer cette requête. 

> ![Manager kms Menu](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_okms_id-optimized-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
![Manager KMS Menu](images/manager_okms_id-optimized-optim-resize.png){.thumbnail}
```

> ### Via l'api OVHcloud

Reviewer: ```suggestion
#### Via les API OVHcloud
```

> Pour lister vos commandes KMS OVHcloud, vous pouvez utiliser l'appel api suivant:

Reviewer: ```suggestion
Pour lister vos commandes KMS OVHcloud, utilisez l'appel API suivant:
```

> [!api]
>
> @api {v2} /okms GET /okms/resource
>

> **Paramètres:**
>

> > - okmsId: L'ID de votre KMS OVHcloud (Okms).

Reviewer: ```suggestion
> - `okmsId`: L'ID de votre KMS OVHcloud (Okms).
```

>

Exemple de retour:
```Shell
[
  {
    "id": "Null",
    "region": "EU_WEST_RBX",
    "kmipEndpoint": "eu-west-rbx.okms.ovh.net:5696",
    "restEndpoint": "https://eu-west-rbx.okms.ovh.net",
    "swaggerEndpoint": "https://swagger-eu-west-rbx.okms.ovh.net",
    "iam": {
      "displayName": "Null",
      "id": "Null",
      "urn": "urn:v1:eu:resource:okms:Null"
    }
  }
]
```

> Vous constatez que vous disposez bien maintenance d'un serveur KMS OVHcloud disponible à mettre en place au sein de votre environnement HPC VMware on OVHcloud.

Reviewer: ```suggestion
Vous disposez désormais d'un serveur KMS OVHcloud à mettre en place au sein de votre environnement HPC VMware on OVHcloud.
```

///

> ## Étape 2 - Activation du KMS OVHcloud (obligatoire)

Reviewer: ```suggestion
### Étape 2 - Activation du KMS OVHcloud (obligatoire)
```

/// details | Comment activer le KMS OVHcloud avec HPC VMware on OVHcloud?

> Afin de valider le KMS OVHloud (Okms) avec Hosted Private Cloud VMware on OVHcloud, vous avez besoin de créer une règle d'ouverture de flux entrant au sein de la gateway OVHcloud pour votre HPC VMware on OVHcloud.

Reviewer: ```suggestion
Pour valider le KMS OVHloud (OKMS) avec Hosted Private Cloud VMware on OVHcloud, créez une règle d'ouverture de flux entrant au sein de la gateway OVHcloud pour votre HPC VMware on OVHcloud.
```

> Cette étape doit être réalisée en premier après commande de votre KMS (OVHcloud). Et avant l'ajout du KMS à vSphere.

Reviewer: ```suggestion
Cette étape doit être réalisée **immédiatement** après commande de votre KMS (OVHcloud) **et avant** l'ajout du KMS à vSphere.
```

> ### Via le control panel OVHcloud

Reviewer: ```suggestion
#### Via l'espace client OVHcloud
```

> ### Étape d'ouverture des flux (obligatoire)

Reviewer: ```suggestion
##### **Ouverture des flux (obligatoire)**
```

> Pour créer ou Importer un service de gestion de clé KMS depuis le control panel Hosted Private Cloud on OVHcloud.

Reviewer: ```suggestion
Pour créer ou Importer un service de gestion de clé KMS, connectez-vous à votre [espace client OVHcloud](/links/manager), puis rendez-vous dans la partie `Hosted Private Cloud`.{.action}. Dans la colonne de gauche, cliquez sur `VMware`.{.action}, puis puis sélectionnez votre PCC concerné. Sur la page qui s'affiche, cliquez sur l'onglet `Sécurité`.{.action}.
```

> 

Reviewer: ```suggestion
```

> Vous devez vous connecter au [control panel OVHcloud](/links/manager).

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> Puis aller dans la section **Sécurité** de votre PCC HPC:

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> - `Hosted Private Cloud > VMware > Votre PCC > Sécurité`{.action}.

Reviewer: ```suggestion
```

> ![Manager Hpc Security KMS](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_hpc_security-opti.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS](images/manager_hpc_security-opti.png){.thumbnail}
```

> Une fois dans `Sécurité`{.action}, allez toute en bas dans la section: `Virtual Machine Encryption Key Management Servers`{.action}.

Reviewer: ```suggestion
Dans l'onglet `Sécurité`{.action}, rendez-vous plus bas dans la section: `Virtual Machine Encryption Key Management Servers`{.action}.
```

> L'étape de l'ajout du KMS depuis le control panel HPC VMware on OVHcloud doit être réalisé en premier après l'achat et la livraison de votre KMS OVHcloud. Afin de permettre l'autorisation des flux au sein des pare-feux OVHcloud.

Reviewer: ```suggestion
L'ajout du KMS depuis l'[espace client OVHcloud](/links/manager) doit être réalisé en **immédiatement** après l'achat et la livraison de votre KMS OVHcloud. Ceci pour permettre l'autorisation des flux au sein des pare-feux OVHcloud.
```

> ![Manager Hpc Security KMS](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_hpc_kms_add-optimized.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS](images/manager_hpc_kms_add-optimized.png){.thumbnail}
```

> Vous pouvez ajouter votre Okms depuis le control panel HPC, en cliquant sur:

Reviewer: ```suggestion
Vous pouvez ajouter votre OKMS depuis le control panel HPC, en cliquant sur le bouton `Ajouter un nouveau serveur KMS`{.action}.
```

> - `Ajouter un nouveau serveur KMS`{.action}

Reviewer: ```suggestion
```

> Une fois la fenêtre ouverte vous vez les champs suivant à remplir :

Reviewer: ```suggestion
Dans la nouvelle fenêtre qui s'affiche, remplissez les formulaires suivants :
```

> - **IP:** Privilégier l'IP car le nom de domaine ne peut pas être ajouté (exemple région strasbourg->137.74.127.152, région roubaix->91.134.128.102).

Reviewer: ```suggestion
- **IP** : Privilégiez l'adresse IP car le nom de domaine ne peut pas être ajouté. Par exemple, utilisez l'adresse IP 137.74.127.152 pour la région de Strasbourg  et l'adresse IP 91.134.128.102 pour la région de Roubaix.
```

> - **Description:** Une description intelligente pour votre Okms.

Reviewer: ```suggestion
- **Description** : Saisissez une description pour votre OKMS.
```

> - **SSL Thumbprint:** La Thumbprint tls/ssl de votre KMS 

Reviewer: ```suggestion
- **SSL Thumbprint** : Renseignez la Thumbprint SSL/TLS de votre KMS 
```

> Pour récupérer l'empreinte tls, la Lancer la commande openssl suivante (adapter votre endpoint okms avec la bonne région eu-west-rbx/sbg dont fait partie votre KMS OVHcloud):

Reviewer: ```suggestion
Pour récupérer l'empreinte TLS, Lancer la commande OpenSSL suivante (adapter votre endpoint OKMS avec la bonne région (par exemple : eu-west-rbx/sbg) dont fait partie votre KMS OVHcloud):
```

```Shell
openssl s_client -connect eu-west-rbx.okms.ovh.net:5696 < /dev/null 2>/dev/null | openssl x509 -fingerprint -noout -in /dev/stdin`{.action})
---
Retour:
 SHA1 Fingerprint=FE:21:E2:DE:B7:51:34:E9:9A:AB:E0:27:FF:1E:42:3A:15:9C:76:47

> ```

Reviewer: ```suggestion
```

```

> Pour récupérer l'IP publique du serveur KMS OVHcloud, lancez un ping par exemple:

Reviewer: ```suggestion
Pour récupérer l'IP publique du serveur KMS OVHcloud, lancez un ping. Par exemple:
```

```Shell
ping eu-west-rbx.okms.ovh.net

> ```

Reviewer: ```suggestion
```

```

> Il s'agit ici de l'IP du KMS de Roubaix et de son empreinte tls. Si vous utilisez Strasbourg ou une autre région faites les changements.

Reviewer: ```suggestion
Il s'agit ici de l'IP du KMS de Roubaix et de son empreinte TLS. Adaptez la commande ci-dessus en fonction de la région où se trouve votre KMS (Strasbourg par exemple).
```

> ![Manager Hpc Security KMS Add 02](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_hpc_kms_add_2-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS Add 02](images/manager_hpc_kms_add_2-optim-resize.png){.thumbnail}
```

> Vous devez bien cocher la case: **"J’ai lu et j’ai compris la documentation VM Encryption et les actions que je vais devoir effectuer de mon côté"**. Avant de passer à l'étape suivante.

Reviewer: ```suggestion
Vérifiez bien que la case: **"J’ai lu et j’ai compris la documentation VM Encryption et les actions que je vais devoir effectuer de mon côté"** est cochée avant de poursuivre.
```

> ![Manager Hpc Security KMS Add 02_2](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_hpc_kms_add_2_2.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS Add 02_2](images/manager_hpc_kms_add_2_2.png){.thumbnail}
```

> Vous devez maintenant attendre que l'ouverture des flux se fasse et passe en **"livré"**.

Reviewer: ```suggestion
Attendez que l'ouverture des flux s'effectue et passe en statut `livré`.
```

> Vous pouvez en attendant verifier vos droits au sein du PCC. En effet, pour manipuler KMS, vous avez besoin de droits supplémentaires:

Reviewer: ```suggestion
En parallèle de l'ouverture des flux, verifiez vos droits au sein du PCC. En effet, pour manipuler KMS, vous aurez besoin de droits supplémentaires.
```

> Dans l'onglet Utilisateurs, cliquez sur: `Modifier`{.action}, l'utilisateur souhaité.

Reviewer: ```suggestion
Pour cela, rendez_vous dans l'onglet `Utilisateurs`{.action} de votre PCC. Dans le tableau qui s'affiche, cliquez sur le bouton `...`{.action} à droite l'utilisateur concerné, puis sur `Modifier`{.action}.
```

> ![Manager Hpc Security KMS Add 02](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_users_edit.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS Add 02](images/manager_users_edit.png){.thumbnail}
```

> Et vérifiez bien que le `management du chiffrement`{.action} est activé.

Reviewer: ```suggestion
Vérifiez que la `gestion du chiffrement`{.action} est activé.
```

> ![Manager Hpc Security KMS Add 02](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_users_edit_encryption.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS Add 02](images/manager_users_edit_encryption.png){.thumbnail}
```

> Si vous utilisez un role IAM au sein de votre PCC faite le pour ce rôle (si vous utilisez le rôle par defaut, il est admin).

Reviewer: ```suggestion
Si vous utilisez un rôle IAM au sein de votre PCC, utilisez-le pour ce rôle. S'il s'agit du rôle par défaut, celui-ci dispose du droit **admin**.
```

> Si vous utilisez une policy avec le role IAM par defaut, vérifiez bien que les ressources PCC lui sont ajoutés, ainsi que les droits KMS et IAM. 

Reviewer: ```suggestion
Si vous utilisez une policy avec le rôle IAM par defaut, vérifiez que les ressources PCC, les droits KMS et IAM lui sont ajoutés. 
```

> Exemple de politique IAM :

Reviewer: ```suggestion
Exemple de politique IAM :

```

> - **Identité:** local user XX -> User local OVHcloud.

Reviewer: ```suggestion
- **Identité** : local user XX (Utilisateur local OVHcloud).
```

> - **User groups:** ADMIN, XXXX-XX-XX/user_iam.

Reviewer: ```suggestion
- **User groups** : ADMIN, XXXX-XX-XX/user_iam.
```

> - **Resources:** pcc-XXX-XXX-XXX-XXX -> PCC XXX.

Reviewer: ```suggestion
- **Resources** : pcc-XXX-XXX-XXX-XXX (référence de votre PCC).
```

> - **Product type:** iam_ressource_type_okms/kmip.

Reviewer: ```suggestion
- **Product type** : iam_ressource_type_okms/kmip.
```

> - **Actions:** vSphere Admin, pccVMware:apiovh:vmEncryption/kms/changeProperties, pccVMware:vSphere:assumeRole?iam-admin -> User vSphere iam-admin, okms:kmip:get, okms:apikms:serviceKey/create etc..

Reviewer: ```suggestion
- **Actions** : vSphere Admin, pccVMware:apiovh:vmEncryption/kms/changeProperties, pccVMware:vSphere:assumeRole?iam-admin -> User vSphere iam-admin, okms:kmip:get, okms:apikms:serviceKey/create etc..
```

> Pour information le **domain id** correspond à l'urn de votre KMS OVHcloud.

Reviewer: ```suggestion
Pour information le **domain ID** correspond à l'URN de votre KMS OVHcloud.
```

> ### Via l'api OVHcloud (optionnel)

Reviewer: ```suggestion
#### Via les API OVHcloud (optionnel)
```

> Si vous avez réalisé l'étape précédente d'ouverture des flux depuis le control panel, vous pouvez passer cette étape depuis l'api.

Reviewer: ```suggestion
Si vous avez déjà réalisé l'ouverture des flux depuis l'[espace client OVHcloud](/links/manager), vous pouvez passer cette étape depuis l'API.
```

> **Étape d'ouverture des flux depuis l'api:**

Reviewer: ```suggestion
**Étape d'ouverture des flux depuis l'API:**
```

> [!api]
>
> @api {v1} /dedicatedCloud POST /dedicatedCloud/{serviceName}/vmEncryption/kms
>

> **Paramètres:**
>

> > - serviceName: La référence de votre PCC, `pcc-XX-XX-XX-XX`.

Reviewer: ```suggestion
> - `serviceName` : Saisissez la référence de votre PCC. Exemple : **pcc-XX-XX-XX-XX**.
```

> > - description: Description de l'Okms.

Reviewer: ```suggestion
> - `description`: Renseignez la description de votre OKMS.
```

> > - ip: L'IP publique de l'Okms.

Reviewer: ```suggestion
> - `ip` : Saisissez l'adresse IP publique de votre OKMS.
```

> > - sslThumbprint: L'empreinte tls de l'Okms.

Reviewer: ```suggestion
> - `sslThumbprint` : Renseignez l'empreinte TLS de votre OKMS.
```

>

> > Copy-past (with the KMS paramètres) :

Reviewer: ```suggestion

Copier-coller (avec les paramètres KMS) :

```

> ```Shell
> {

> >  "description": "Okms demo",

Reviewer: ```suggestion
>  "description": "OKMS descrpition",
```

>  "ip": "Null",
>  "sslThumbprint": "Null"
> }
> ```

> Pour récupérer l'empreinte tls du KMS, lancer la commande **openssl** suivante:

Reviewer: ```suggestion
Pour récupérer l'empreinte TLS du KMS, lancer la commande **OpenSSL** suivante en adaptant la commande à la région où se trouve votre KMS:
```

```Shell
openssl s_client -connect eu-west-rbx.okms.ovh.net:5696 < /dev/null 2>/dev/null | openssl x509 -fingerprint -noout -in /dev/stdin
---
Retour:
 SHA1 Fingerprint=FE:21:E2:DE:B7:51:34:E9:9A:AB:E0:27:FF:1E:42:3A:15:9C:76:47
```

> Pour récupérer l'IP publique du KMS OVHcloud, effectuez un ping:

Reviewer: ```suggestion
Pour récupérer l'IP publique du KMS OVHcloud, effectuez un ping en adaptant la commande à la région où se trouve votre KMS:
```

```Shell
ping eu-west-rbx.okms.ovh.net
```

Pour mettre à jour votre KMS avec un KMS OVHcloud:

> [!api]
>
> @api {v1} /dedicatedcloud  POST /dedicatedCloud/{serviceName}/vmEncryption/kms/{kmsId}/changeProperties
>

> **Paramètres:**
>

> > - serviceName: Le domaine pcc (pcc-XXX-XXX-XXX).

Reviewer: ```suggestion
> - `serviceName` : Renseignez le nom de votre PCC. Exemple : pcc-XX-XX-XX-XX.
```

> > - kmsId: L'id de votre serveur Okms (exemple: 350)

Reviewer: ```suggestion
> - `kmsId` : Saisissez l'ID de votre serveur OKMS. (Exemple: 350)
```

>

> Retour:

Reviewer: ```suggestion
Après exécution de l'API, vous devriez retrouvez en réponse le résultat suivant:
```

```Shell
{
  "kmsId": 350,
  "kmsTcpPort": 5696,
  "sslThumbprint": "Null",

>   "description": "Okms demo",

Reviewer: ```suggestion
  "description": "OKMS description",
```

  "state": "delivered",
  "ip": "Null"
}
```

> Vous devez maintenant attendre (statut: updating) que l'ouverture des flux se fasse et passe le statut passe en: `livré`{.action} (optionnel).

Reviewer: ```suggestion
Attendez (statut: updating) que l'ouverture des flux s'effectue et que le statut passe en état `livré` (optionnel).
```

> ![Manager Hpc Security KMS Delivered](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_hpc_kms_add_delivered.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC Security KMS Delivered](images/manager_hpc_kms_add_delivered.png){.thumbnail}
```

///

> ## Étape 3 - Ajout du KMS à vSphere/vCenter (obligatoire)

Reviewer: ```suggestion
### Étape 3 - Ajout du KMS à vSphere/vCenter (obligatoire)
```

/// details | Comment ajouter le KMS OVHcloud dans vSphere/vCenter?

> ### Via le control panel OVHcloud

Reviewer: ```suggestion
#### Via l'espace client OVHcloud
```

> ![Manager Hpc General Information Web Interface](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_hpc_vsphere.png){.thumbnail}

Reviewer: ```suggestion
![Manager HPC General Information Web Interface](images/manager_hpc_vsphere.png){.thumbnail}
```

> #### 1. Ajouter un fournisseur de clé KMS

Reviewer: ```suggestion
##### 1. Ajouter un fournisseur de clé KMS
```

Pour que vCenter puisse truster votre serveur KMS OVHcloud, nous avons besoin d'accéder à la console vSphere de votre PCC HPC VMware on OVHcloud.

> Pour accéder à l'url de connexion vSphere depuis un navigateur web (avec de preference une base navigateur chromium).

Reviewer: ```suggestion
```

Reviewer: This sentece seems to be useless because you didn't explain how to acces on vSphere URL to a browser and you explain juste after this sentence how to go to the console from the OVH control panel.

> 

Reviewer: ```suggestion
```

> Vous pouvez y accéder depuis le control panel OVHcloud, en cliquant sur:

Reviewer: ```suggestion
Pour cela, connectez-vous à votre [espace client OVHcloud](/links/manager), puis rendez-vous dans la partie `Hosted Private Cloud`{.action}. Dans la colonne de gauche, cliquez sur  `VMware`{.action} et sélectionnez le PCC concerné.
```

> - `Hosted Private Cloud > VMware > Votre PCC > Manager interface > Web interface`{.action}.

Reviewer: ```suggestion

Sur la page qui s'affiche et dans l'encadré `Informations générales`, descendez jusqu'à retrouver la mention `Interfaces de gestion`{.action}, puis cliquez en dessous sur `Interface web`{.action}.
```

> Une fois que vous avez cliqué sur Web interface, cliquez sur `vSphere HTML Client`{.action}.

Reviewer: ```suggestion
Sur la nouvelle page qui apparaît, cliquez sur le carré intitulé `vSphere HTML Client`{.action}.
```

> ![Manager Web Interface](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/manager_web_interface_pcc.png){.thumbnail}

Reviewer: ```suggestion
![Manager Web Interface](images/manager_web_interface_pcc.png){.thumbnail}
```

> Vous êtes maintenant sur l'url de votre pcc, par exemple: 

Reviewer: ```suggestion
Vous êtes désormais sur la page de login ou d'accueil de votre PCC. L'URL en haut de votre navigateur doit ressembler à ceci: `<https://pcc-x.x.x.x.ovh.de/ui/login>`.
```

> - `<https://pcc-x.x.x.x.ovh.de/ui/>`{.action}

Reviewer: ```suggestion
```

> Connectez-vous avec un utilisateur local ou avec un utilisateur IAM selon les droits que vous avez mis en place au sein de votre compte OVHcloud et de votre PCC HPC on OVHcloud.

Reviewer: ```suggestion
Connectez-vous avec un **utilisateur local** ou avec **un utilisateur IAM** selon les droits que vous avez mis en place au sein de votre [espace client OVHcloud](/links/manager) et de votre PCC HPC on OVHcloud.
```

> Vous êtes maintenant logué au sein de votre Hosted Private Cloud vSphere on OVHcloud.

Reviewer: ```suggestion
Vous êtes maintenant connecté au sein de votre Hosted Private Cloud vSphere on OVHcloud.
```

> Pour accéder à la gestion des serveurs KMS dans vSphere, cliquer sur: `Configurer`{.action} depuis votre **pcc-XXX-XXX-XXX-XXX**.

Reviewer: ```suggestion
Pour accéder à la gestion des serveurs KMS depuis vSphere, cliquez sur  `Configurer`{.action} depuis votre **pcc-XXX-XXX-XXX-XXX**. Rendez-vous dans la section `Sécurité`{.action}, puis cliquez sur `Fournisseurs de clés`{.action}. Sur la page qui apparaît, cliquez sur le bouton `Ajouter`{.action}, puis sur `Ajouter un fournisseur de clé standard`{.action}.
```

> 

Reviewer: ```suggestion
```

> Puis, allez dans la section **Sécurité**: `Fournisseurs de clés`{.action}.

Reviewer: ```suggestion
```

> Il vous faut maintenant, cliquez sur: `Ajouter > Ajouter un fournisseur de clé standard`{.action}.

Reviewer: ```suggestion
```

> ![KMS Key Provider](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/kms_key_provider.png){.thumbnail}

Reviewer: ```suggestion
![KMS Key Provider](images/kms_key_provider.png){.thumbnail}
```

> Une fois que vous avez sélectionné l'option pour ajouter un Key Provider, une fenêtre ou un formulaire s'ouvrira pour saisir les détails du Key Provider que vous souhaitez ajouter. Cela peut inclure des informations telles que l'adresse IP ou le nom de domaine (DNS) du serveur Okms, ainsi que le port utilisé (5696).

Reviewer: ```suggestion
Une fois  l'option sélectionnée pour ajouter un Key Provider, une fenêtre ou un formulaire s'ouvre pour saisir les détails du **Key Provider** que vous souhaitez ajouter. Cela peut inclure des informations telles que l'adresse IP ou le nom de domaine (DNS) du serveur OKMS, mais aussi le port utilisé (5696).
```

> Les noms de domaines et le port (KMIP) ne changent pas.

Reviewer: ```suggestion
Les noms de domaine et le port (KMIP) ne changent pas.
```

> ![KMS Key Provider](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_add_standard_key_provider.png){.thumbnail}

Reviewer: ```suggestion
![KMS Key Provider](images/okms_vsphere_add_standard_key_provider.png){.thumbnail}
```

> Vous avez ces champs disponibles:

Reviewer: ```suggestion
Vous y retrouvez les champs suivants:
```

> | Champ       | Input                                                   | Description                                                                  |

Reviewer: ```suggestion
| Champ       | Valeur                                                   | Description                                                                  |
```

|-------------|---------------------------------------------------------|------------------------------------------------------------------------------|

> | **Name**    |                                                         | - Un champ qui permet de nommer votre cluster au sein de vCenter.            |

Reviewer: ```suggestion
| **Name**    |                                                         | Permet de nommer votre cluster au sein de vCenter.            |
```

> | **KMS**     |                                                         | - Le nom qui apparaitra au sein de Vsphere pour votre Okms.                  |

Reviewer: ```suggestion
| **KMS**     |                                                         | Nom qui apparaitra au sein de Vsphere pour votre OKMS.                  |
```

> | **Address** | eu-west-rbx.okms.ovh.net <br/> eu-west-sbg.okms.ovh.net | - L'endpoint du serveur Okms, privilégier le nom de domaine plutôt que l'IP. |

Reviewer: ```suggestion
| **Address** | eu-west-rbx.okms.ovh.net <br/> eu-west-sbg.okms.ovh.net | **Endpoint** du serveur OKMS. Privilégiez le nom de domaine plutôt que l'IP. |
```

Reviewer: Really ? because since the begining of the guide you told that it's preferable to use IP.

Author: There is the first process of opening flow in the PCC which can only be done with IP (end of step 2). And there is after the process of adding the KMS in vSphere web console where you can do both (recommend using DNS) because if IP changes it need to be updated (step 3). Not with DNS.

> | **Port**    | 5696                                                    | - Port utilisé par KMIP (ne change pas).

Reviewer: ```suggestion
| **Port**    | 5696                                                    | Port utilisé par KMIP (ne change pas).
```

> Attendez que vSphere établisse la connexion avec le Key Provider que vous avez ajouté. Vous devriez voir une indication ou un message confirmant que la connexion a été établie avec succès.

Reviewer: ```suggestion
Patientez le temps que vSphere établisse la connexion avec le Key Provider que vous venez d'ajouter. Vous devriez voir apparaitre un message confirmant que la connexion a été établie avec succès.
```

> ### KMS approuve vCenter + vCenter approuve KMS

Reviewer: ```suggestion
#### Approbation du serveur KMS par vCenter et inversement
```

> Afin de communiquer avec votre KMS, il est nécessaire de créer un certificat d'accès. Celui-ci sera utilisé pour toute interaction avec le KMS OVHcloud et vSphere/vCenter, que ce soit pour créer des clés de chiffrement ou effectuer des opérations avec ceux-ci.

Reviewer: ```suggestion
Afin de communiquer avec votre KMS, vous devez créer un certificat d'accès. Celui-ci sera utilisé pour l'ensemble des interactions avec le KMS OVHcloud et vSphere/vCenter, que ce soit pour créer des clés de chiffrement ou effectuer des opérations avec ceux-ci.
```

Chaque certificat contient une [identité OVHcloud](/pages/manage_and_operate/iam/identities-management) permettant de calculer les droits d'accès via l'[IAM OVHcloud](/pages/account_and_service_management/account_information/iam-policy-ui).

> Il est possible de générer ce certificat en laissant OVHcloud générer la clé privée, ou en fournissant votre propre clé de sécurité privée via une **"Certificate Signing Request (CSR)"**.

Reviewer: ```suggestion
Vous pouvez générer ce certificat en laissant OVHcloud générer la clé privée, ou en fournissant votre propre clé de sécurité privée via une **C**ertificate **S**igning **R**equest (**CSR**).
```

> ### Sans CSR (without CSR)

Reviewer: ```suggestion
##### **Sans CSR (without CSR)**
```

> Vous pouvez créer un certificat d'accès et une clé privée avec l'appel api suivant: `Sans CSR (without CSR provided)`{.action}

Reviewer: ```suggestion
Vous pouvez créer un certificat d'accès et une clé privée avec l'appel API suivant: `Sans CSR (without CSR provided)`{.action}
```

> [!api]
>
> @api {v2} /okms POST /okms/resource/{okmsId}/credential
>

> **Paramètres:**
>

> > - okmsId: L'ID de votre KMS OVHcloud (Okms).

Reviewer: ```suggestion
> - `okmsId`: L'ID de votre KMS OVHcloud (OKMS).
```

> - Avec/Sans CSR Provided.
>

Il est nécessaire d'indiquer les informations suivantes :

> - **name:** Le nom du certificat (obligatoire).

Reviewer: ```suggestion
- `name` : nom du certificat (obligatoire).
```

> - **identityURNs:** L'identité OVHcloud sous format d'Urn qui sera fourni à IAM pour le calcul des droits d'accès (obligatoire).

Reviewer: ```suggestion
- `identityURNs` : Identité OVHcloud sous format d'URN qui sera fourni à IAM pour le calcul des droits d'accès (obligatoire).
```

> - **description:** Description du certificat (optionnel).

Reviewer: ```suggestion
- `description` : Description du certificat (optionnel).
```

> - **validity:** Durée de validité du certificat en jours - 365 jours par défaut (optionnel).

Reviewer: ```suggestion
- `validity` : Durée de validité du certificat en jours - 365 jours par défaut (optionnel).
```

> **Voici un exemple de création de certificat:**

Reviewer: ```suggestion
**Voici un exemple de création de certificat:**

```

```Shell
{
 "description": "My user reader credential",
 "identityURNs": [
 "urn:v1:labeu:identity:account:user:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/user" // L'utilisateur avec lequel vous vous connectez à vSphere au sein de votre PCC, il doit être admin ou avoir Encryption management activé au sein du PCC pour chiffrer des VM. 
 "urn:v1:eu:identity:group:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/group" // Le groupe avec lequel vous accédez à vos PCC, pour activer le chiffrement sur les VM vous devez être admin ou activer Encryption management.
 ],
 "name": "my_reader", // Nom du reader.
 "validity": 30 // Par defaut 365 jours.
}
```

> L'api retourne ensuite l'état de création du certificat:

Reviewer: ```suggestion
L'API retourne ensuite l'état de création du certificat:

```

```Shell
{
  "id": "1bc7830b-9c8e-4b45-af51-00a3543ba16c",
  "name": "user",
  "description": "XXX",
  "identityURNs": [
    "urn:v1:eu:identity:user:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/user",
    "urn:v1:eu:identity:group:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/group"
  ],
  "status": "CREATING",
  "fromCSR": false,
  "privateKeyPEM": "-----BEGIN EC PRIVATE KEY-----\nXXcCAQEEIEs+RwZG0JLs5bmzxVxZtli15RLGU/cXXHtoVoOrwDzuoAoGCCqGSM49\nAwEHoUQDQgAExC+jRLpucMcNUcXXhxHNMA78EjrmmKnMgeEaRzAvX2dY4GVap5lU\nM76nD1WRnNvceYHZUOxGcPDCuXXKNV1/8A==\n-----END EC PRIVATE KEY-----\n",
  "createdAt": "2024-07-02T15:54:18.390593+02:00",
  "expiredAt": "2024-08-01T15:54:18.390591+02:00"
}
```

> Copier la valeur du champ (exemple ci-dessus): `privateKeyPEM`{.action}, puis coller là dans un fichier: `domain.key`{.action}. Uniquement `-----BEGIN EC YOUR_PRIVATE_KEY-----\n...\n-----END EC YOUR_PRIVATE_KEY-----`{.action}.

Reviewer: ```suggestion
Copiez la valeur du champ (exemple ci-dessus): `privateKeyPEM`{.action}, puis collez-la dans un fichier: `domain.key`{.action}. Uniquement de `-----BEGIN EC YOUR_PRIVATE_KEY-----\n` à `\n-----END EC YOUR_PRIVATE_KEY-----`{.action}.
```

> Enregistrer ce fichier afin de le réutiliser pour que **KMS trust vCenter**: `Make KMS Trust vCenter`{.action}.

Reviewer: ```suggestion
Enregistrez ce fichier afin de le réutiliser pour le **KMS trust vCenter** ultérieurement.
```

> Puis copier aussi l'id généré et lancer l'appel api suivant avec l'id et l'okmsId :

Reviewer: ```suggestion
Copiez également l'ID généré et lancer l'appel API suivant avec l'ID et l'okmsId :
```

> [!Warning]
>

> > La clé privée ne sera plus accessible par la suite. En cas de perte, il sera nécessaire de re-générer un certificat.

Reviewer: ```suggestion
> La clé privée ne sera plus accessible par la suite. En cas de perte, vous devrez re-générer un certificat.
```

>

> [!api]
>
> @api {v2} /okms GET /okms/resource/{okmsId}/credential/{credentialId}
>

> **Paramètres:**
>

> > - id: L'id généré par le POST /okms/resource/{okmsId}/credential, exemple (2bc7830b-9c8e-4b45-af51-00a3543ba16c).

Reviewer: ```suggestion
> - `id` : ID généré par le **POST /okms/resource/{okmsId}/credential**. Exemple : 2bc7830b-9c8e-4b45-af51-00a3543ba16c.
```

> > - okmsId: L'id de votre KMS OVHcloud (Okms).

Reviewer: ```suggestion
> - `okmsId:`  ID de votre KMS OVHcloud (OKMS).
```

    "urn:v1:eu:identity:group:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/group" // Le groupe OVHcloud avec lequel vous accédez à vos PCC, pour activer le chiffrement sur les VM vous devez être admin ou activer Encryption management.
  ],
  "status": "READY",
  "fromCSR": false,
  "certificatePEM": "-----BEGIN CERTIFICATE-----\nMXXBqTCCAU+gAwIBAgIQBu8UoHY1TxeHSKzPrb1iAzAKBggqhkjOPXXDAjAVMRMw\nEQYDVQQDEwpDQ01Vc2Vyc0NBMB4XDTI0MDcwMjEzNTQxOFoXDTI0MDgwMTEzNTQx\nOFowLzEtMCsGA1UEAxMkZTkwYzU4NDEtZjM5ZSXXOTgzLTg1OTYtNjJmZjBjNTM4\nYjZiMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAExC+jRLpucMcNUc84hxHNMA78\nEjrmmKnMgeEaRzAvX2dY4GVap5lUM76nD1WRnNvceYHZUOxGcPDCu33KNV1/8KNn\nMGUwTgYDVR0RAQH/BEQwQqBABgorBgEEAYI3FAIDoDIMMG9rbXMuZG9tYWluOmU5\nMGM1ODQxLWYzOWUtNDk4My04NTk2LTYyZmYwYzUzOGI2YjATBgNVHSUEDDAKBggr\nBgEFBQcDAjAKBggqhkjOPQQDAgNIADBFAiBD9GOB4c+7IVQzlJBCP8TZ9O3M4pHS\nVJTyEYxld5eXYQIhAL/1fe5qGBZib4vlaPss6zgJoentyMUq+zTCMUn5wXcP\n-----END CERTIFICATE-----",
  "createdAt": "2024-07-02T15:54:18.390593+02:00",
  "expiredAt": "2024-08-01T15:54:18.390591+02:00"
}
```

> Copier la valeur du champ `"certificatePEM"`{.action} `"-----BEGIN CERTIFICATE-----\nYOUR_CERTIFICATE\n-----END CERTIFICATE-----"`{.action} puis coller là (printf) dans un fichier `client.pem`{.action}.

Reviewer: ```suggestion
Copiez la valeur du champ `"certificatePEM"`{.action} `"-----BEGIN CERTIFICATE-----\nYOUR_CERTIFICATE\n-----END CERTIFICATE-----"`{.action}, puis collez-la (printf) dans un fichier `client.pem`{.action}.
```

> ### Avec CSR

Reviewer: ```suggestion
##### **Avec CSR**
```

Suivez les mêmes étapes que ci-dessus, mais avec le champ `With CSR provided`{.action} de l'appel api suivant:

> [!api]
>
> @api {v2} /okms POST /okms/resource/{okmsId}/credential
>

> **Paramètres:**
>

> > - okmsId: L'ID de votre KMS OVHcloud (Okms).

Reviewer: ```suggestion
> - `okmsId`: ID de votre KMS OVHcloud (OKMS).
```

  "description": "My reader credential",
  "identityURNs": [
    "urn:v1:eu:identity:user:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/user",
    "urn:v1:eu:identity:group:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/user"
  ],
  "name": "reader name",
  "validity": 365
}
```

> Coller la valeur du champ `"csr"`{.action}: `"-----BEGIN CERTIFICATE REQUEST-----\nYOUR_CERTIFICATE\n-----END CERTIFICATE REQUEST-----"`{.action} récupéré de vSphere on OVHcloud.

Reviewer: ```suggestion

Collez la valeur du champ `"csr"`{.action} : `"-----BEGIN CERTIFICATE REQUEST-----\nYOUR_CERTIFICATE\n-----END CERTIFICATE REQUEST-----"`{.action} récupérée de vSphere on OVHcloud.
```

> > [!tabs]

Reviewer: Using a "tab" should'nt be pertinent. Maybe try to search another solution to expose these elements.

> >>

Reviewer: ```suggestion
```

> >> ### vCenter Trust KMS

Reviewer: ```suggestion
> vCenter Trust KMS
```

>>

> >> **Faire que vCenter approuve KMS**

Reviewer: ```suggestion
>> **Approbation de vCenter pour le KMS**
```

>>

> >> ![Trust KMS server with or without CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_kms_vsphere_configuration/images/trust_kms..png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with or without CSR](images/trust_kms..png){.thumbnail}
```

>>

> >> D'abord, vous devez approuver le kMS au sein de vCenter. Pour le faire, cliquez sur: `Approuver le KMS`{.action}.

Reviewer: ```suggestion
>> Cliquez sur `Approuver le KMS`{.action} pour approuver le KMS au sein de vCenter.
```

>>

> >> En principe, vCenter va télécharger automatiquement le certificat public KMS. S'il ne le fait pas, vous pouvez le faire manuellement en cliquant sur: `Établir une relation de confiannce > vCenter approuve Kms > Télécharger le vertificat KMS`{.action}.

Reviewer: ```suggestion
>> En principe, vCenter télécharge automatiquement le certificat public KMS. S'il ne le fait pas, faites-le manuellement en cliquant sur `Établir une relation de confiannce` > vCenter approuve Kms > Télécharger le certificat KMS`{.action}.
```

Reviewer: I don't know where I go. I didn't find the menu `Établir une relation de confiannce`. Where is it ? please explain with more precisions where this menu is.

>>

> >> L'approbation entre vCenter et KMS peut être conflictuelle, attendez quelques minutes avant de recommencer ou rafraichissez votre page vSphere avec le bouton en haut (flèche en forme de rond).

Reviewer: ```suggestion
>> L'approbation entre vCenter et KMS peut être conflictuelle, patientez quelques minutes avant de recommencer ou rafraichissez votre page vSphere avec le bouton en forme de flèche circulaire situé en haut et légèrement à droite.
```

>>

> >> ![KMS Key Provider with or without CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_vcenter_trust_kms_1-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
>> ![KMS Key Provider with or without CSR](images/okms_vsphere_vcenter_trust_kms_1-optim-resize.png){.thumbnail}
```

>>

> >> ![KMS Key Provider with or without CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_vcenter_trust_kms-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
>> ![KMS Key Provider with or without CSR](images/okms_vsphere_vcenter_trust_kms-optim-resize.png){.thumbnail}
```

>>

> >> Après ça, il faut que le **KMS approuve vCenter**.

Reviewer: ```suggestion
>> Ensuite, il faut que le **KMS approuve vCenter**.
```

>> 

> >> Sélectionner votre **Fournisseurs de clés** KMS (Okms) que vous venez d'ajouter et cliquer sur le bouton: `Approuver l'instance de VCENTER`{.action}.

Reviewer: ```suggestion
>> Sélectionnez votre **Fournisseurs de clés** KMS (OKMS) que vous venez d'ajouter et cliquez sur le bouton `Approuver l'instance de VCENTER`{.action}.
```

>>

> >> ![Trust KMS server with or without CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_kms_vsphere_configuration/images/trust_kms..png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with or without CSR](images/trust_kms.png){.thumbnail}
```

>>

> >> Nous recommandons la méthode avec CSR (plus sécurisé): `Nouvelle demande de signature de certificat (CSR)`{.action), mais libre à vous de choisir celle qui vous convient le mieux et qui est compatible avec votre façon de faire.

Reviewer: ```suggestion
>> Nous recommandons la méthode avec CSR (plus sécurisée) en cliquant sur `Nouvelle demande de signature de certificat (CSR)`{.action). Cependant, libre à vous de choisir celle qui vous convient le mieux et qui est compatible avec votre façon de faire.
```

>>
>> Pour plus d'information sur les avantages et inconvénients de l'utilisation d'une CSR, lisez la documentation [KMS](/pages/manage_and_operate/kms/quick-start).
>>

> >> ### KMS trust vCenter

Reviewer: ```suggestion
> KMS trust vCenter
```

>>
>> **Établir une relation de confiance entre vCenter et KMS:**
>>

> >> Si en cliquant sur Approuver KMS vCenter a bien téléchargé le certificat public, cette étape est optionnel.

Reviewer: ```suggestion
>> > [!primary] 
>> >
>> > Si, en cliquant sur `Approuver KMS`, vCenter a bien téléchargé le certificat public, cette étape est optionnelle.
```

>>

> >> Car avant de générer votre CSR et/ou votre Certificat KMS et clé privée, vous devez verifier que vCenter à bien télécharger le certificat public du serveur KMS OVHcloud.

Reviewer: ```suggestion
>> Effectivement, avant de générer votre CSR et/ou votre Certificat KMS, puis votre clé privée, vous devez verifier que vCenter à bien téléchargé le certificat public du serveur KMS OVHcloud.
```

>>

> >> Pour ça vous pouvez le récupérer depuis les endpoint fournis avec ce script par exemple ou une commande openssl:

Reviewer: ```suggestion
>> Si besoin, vous pouvez le récupérer depuis les endpoints fournis avec ce script ou une commande OpenSSL:
```

>>
>> ```Shell
>> openssl s_client -showcerts -connect eu-west-rbx.okms.ovh.net:443 </dev/null 2>/dev/null|openssl x509 -outform PEM | python3 -c "
>> import sys
>> import json
>> body = {}
>> body['cert'] = sys.stdin.read()
>> json.dump(body, sys.stdout)
>> " | python3 -c "
>> import sys
>> import json
>> body = json.load(sys.stdin)
>> print(body['cert'])
>> " | openssl x509 -text; echo $?
>> ```

> >> Copiez le retour depuis `-----BEGIN CERTIFICATE----- XXX -----END CERTIFICATE-----`{.action} dans vSphere.

Reviewer: ```suggestion
>>
>> Copiez le retour depuis `-----BEGIN CERTIFICATE----- XXX -----END CERTIFICATE-----`{.action} dans vSphere.
```

>> 

> >> Il faut cliquer sur: `Établir une realtion de confiance`{.action}.

Reviewer: ```suggestion
>> Cliquez ensuite sur `Établir une relation de confiance`{.action}.
```

Reviewer: please explain where this button is.

>> 

> >> Puis cliquez sur: `Télécharger le certificat KMS`{.action}.

Reviewer: ```suggestion
>> Puis cliquez sur `Télécharger le certificat KMS`{.action}.
```

>> 

> >> Pour finir, coller le certificat public KMS.

Reviewer: ```suggestion
>> Pour terminer, collez le certificat public KMS.
```

>>

> >> ![KMS Key Provider](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_upload_kms_cert-ptim-resize.png){.thumbnail}

Reviewer: ```suggestion
>> ![KMS Key Provider](okms_vsphere_upload_kms_cert-ptim-resize.png){.thumbnail}
```

>> 

> >> ![KMS Key Provider](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_upload_kms_cert_2-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
>> ![KMS Key Provider](images/okms_vsphere_upload_kms_cert_2-optim-resize.png){.thumbnail}
```

>>

> > **Certificat KMS et clé privée (without CSR)**

Reviewer: ```suggestion
>> **Certificat KMS et clé privée (without CSR)**
```

Reviewer: Is it always into the part "KMS trust vCenter" ? It shouldn't be very clear on my side. Maybe using a "tab" to explain these notions is'nt the best method.

>>

> >> Il vous faut cliquez sur: `Faire que KMS approuve vCenter`{.action}

Reviewer: ```suggestion
>> Pour cela, cliquez sur `Faire que KMS approuve vCenter`{.action}.
```

Reviewer: Where is this button ? Could you please explain with more precisions how to acces to this button from vSphere.

>>
>> Puis, sélectionnez: `KMS Certificate and private key to vCenter.`{.action}.
>> 

> >> Pour générer puis signé votre certificat public et la clé privée, lancez les appels api suivants:

Reviewer: ```suggestion
>> Pour générer, puis signer votre certificat public et la clé privée, lancez les appels API suivants:
```

>>
>> [!api]
>>
>> @api {v2} /okms POST /okms/resource/{okmsId}/credential
>>
>> **Paramètre:**
>> 
>>**Without CSR**
>>
>> ```Shell
>> {
>> "description": "My reader credential",
>> "identityURNs": [
>> "urn:v1:eu:identity:user:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/user",
>> "urn:v1:eu:identity:group:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/group"
>> ],
>> "name": "reader",
>> "validity": 365 (exemple)
>> }
>> ```
>>

> >> Récupérez le **credential id** du post ci-dessus, ainsi que la clé privée générée **privateKeyPEM**. Puis collez là dans un fichier **XXX.pem** afin de pouvoir la télécharger (upload) dans vSphere/vCenter.

Reviewer: ```suggestion
>> Récupérez le **credential id** du post ci-dessus, ainsi que la clé privée générée **privateKeyPEM**. Collez-la ensuite dans un fichier **XXX.pem** pour la télécharger (upload) dans vSphere/vCenter.
```

>>

> >> Puis lancer le get suivant pour signer vos clés:

Reviewer: ```suggestion
>> Lancez ensuite l'API suivante pour signer vos clés:
```

>> 
>> [!api]
>>
>> @api {v2} /okms GET /okms/resource/{okmsId}/credential/{credentialId}
>>
>> **Paramètre:**

> >> - **credentialId:** L'id de la requête d'accès à votre Okms.

Reviewer: ```suggestion
>> - `credentialId` : ID de la requête d'accès à votre OKMS.
```

> >> - **okmsId:** L'id de la commande de votre Okms.

Reviewer: ```suggestion
>> - `okmsId` : ID de la commande de votre OKMS.
```

>>

> >> ![Trust KMS server without CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_kms_vsphere_configuration/images/kms_trust_vcenter..png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server without CSR](images/kms_trust_vcenter..png){.thumbnail}
```

>>

> >> ![Trust KMS server without CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_kms_vsphere_configuration/images/kms_trust_vcenter_2..png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server without CSR](images/kms_trust_vcenter_2..png){.thumbnail}
```

>>

> >> Afin d'utiliser la clé signée dans vSphere, vous devez le formater avec cette commande:

Reviewer: ```suggestion
>> Afin d'utiliser la clé signée dans vSphere, formatez-la avec cette commande:
```

>>
>> ```Shell
>> # Clé signé (PEM):
>> awk '{gsub(/\\n/,"\n")}1' cle.pem
>> 
>> # Certifiat signé (CRT):
>> awk '{gsub(/\\n/,"\n")}1' cert.crt
>> ```
>> 

> >> Le format n'a pas de réelle importance, car il n'est pas utilisé au sein de openssl à ce stade.

Reviewer: ```suggestion
>> Le format n'a pas de réelle importance car il n'est pas utilisé au sein d'OpenSSL à ce stade.
```

>> 

> >> Relancer cette commande awk avec le certificat signé, puis copiez-le.

Reviewer: ```suggestion
>> Relancez cette commande awk avec le certificat signé, puis copiez-le.
```

>>

> >> Et pour finir, collez le dans vSphere, puis cliquez sur: `Etablir une relation de confiance`{.action}.

Reviewer: ```suggestion
>> Pour terminer, collez-le dans vSphere, puis cliquez sur `Etablir une relation de confiance`{.action}.
```

>>

> > **Nouvelle demande de signature de certificat (avec CSR/with CSR)**

Reviewer: ```suggestion
>> **Nouvelle demande de signature de certificat (avec CSR/with CSR)**
```

Reviewer: Same remark than before. Is it always into the part KMS trust vSphere ?

>>

> >> Une fois votre KMS commandé, vCenter approuvé avec KMS, vous pouvez lancer la génération du `CSR`{.action} afin que KMS approuve vCenter et signe le CSR.

Reviewer: ```suggestion
>> Une fois que votre KMS est commandé et que vCenter approuve le KMS, lancez la génération du `CSR`{.action} afin que KMS approuve vCenter et signe le CSR.
```

>>

> >> Il faut alors, cliquer sur: `Faire que KMS approuve vCenter`{.action}.

Reviewer: ```suggestion
>> Cliquez sur `Faire que KMS approuve vCenter`{.action}.
```

>>

> >> Sélectionnez donc: `Nouvelle demande de signature de certificat (CSR)`{.action}. Puis Copiez ou téléchargez le CSR ci-dessous et mettez-le à la disposition du KMS OVHcloud depuis l'api v2 /okms et demandez à ce dernier de signer le certificat.

Reviewer: ```suggestion
>> Sélectionnez `Nouvelle demande de signature de certificat (CSR)`{.action}, puis copiez ou téléchargez le CSR ci-dessous. Mettez-le à disposition du KMS OVHcloud depuis l'API v2 /okms et demandez à ce dernier de signer le certificat.
```

>>

> >> La signature ce fait avec l'appel api suivant: 

Reviewer: ```suggestion
>> La signature ce fait avec l'appel API suivant: 
```

>> "description": "My user reader credential",
>> "identityURNs": [
>> "urn:v1:eu:identity:user:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/user",
>> "urn:v1:eu:identity:group:<<PASTE_YOUR_NICHANDLE_HERE>>-ovh/group"
>> ],
>> "name": "user",
>> "validity": 365
>> } 
>> ```
>>

> >> L'approbation ne sera pas établie une fois que vous aurez terminé cet Assistant. Rendez-vous dans le KMS pour télécharger le CSR, faites signer le certificat par le KMS et téléchargez-le sur le vCenter pour établir la confiance.

Reviewer: ```suggestion
>> L'approbation ne sera pas établie une fois que vous aurez terminé cet Assistant. Rendez-vous dans le KMS pour télécharger le CSR, faites signer le certificat par le KMS et téléchargez-le sur le vCenter pour établir la relation de confiance.
```

>>

> >> Attention votre CSR doit être compatible en format json afin de l'utiliser dans l'api v2 OVHcloud.

Reviewer: ```suggestion
>> Attention votre CSR doit être compatible en format json afin de l'utiliser dans l'API v2 OVHcloud.
```

>>

> >> ![Trust KMS server with CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_kms_vsphere_configuration/images/kms_trust_vcenter_csr..png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with CSR](images/kms_trust_vcenter_csr..png){.thumbnail}
```

>>

> >> ![Trust KMS server with CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_kms_vsphere_configuration/images/kms_trust_vcenter_csr_2..png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with CSR](images/kms_trust_vcenter_csr_2..png){.thumbnail}
```

>>

> >> Copier votre CSR dans un fichier `csr.pem`{.action} afin de pouvoir le faire signer par le KMS.

Reviewer: ```suggestion
>> Copiez votre CSR dans un fichier `csr.pem`{.action} pour le faire signer par le KMS.
```

>>

> >> Pour formater le CSR, lancer la commande awk ci-dessous:

Reviewer: ```suggestion
>> Pour formater le CSR, lancer la commande awk ci-dessous:
>>
```

>> ```Shell
>> awk '{printf "%s\\n", $0}' csr.pem
>> ```

> >> Et coller le retour de cette commande dans l'appel api plus haut, pour faire signer votre CSR auprès du KMS.

Reviewer: ```suggestion
>> Collez le retour de cette commande dans l'appel API plus haut, pour faire signer votre CSR auprès du KMS.
```

>>
>> Il ne vous reste plus qu'à uploader le CSR signé.
>> 

> >> Afin de signer le CSR, cliquez sur: `ESTABLISH TRUST > Upload Signed CSR Certificate`{.action}.

Reviewer: ```suggestion
>> Pour signer le CSR, cliquez sur `ESTABLISH TRUST`{.action}, puis sur `Upload Signed CSR Certificate`{.action}.
```

>> 

> >> ![Trust KMS server with CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_upload_signed_csr-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with CSR](images/okms_vsphere_upload_signed_csr-optim-resize.png){.thumbnail}
```

>>

> >> Vous devez maintenant copier le CSR signé de l'appel api suivant:

Reviewer: ```suggestion
>> Copiez maintenant le CSR signé de l'appel API suivant:
```

>> 
>> [!api]
>>
>> @api v2 /okms GET /okms/resource/{okmsId}/credential/{credentialId}
>> 
>> **Paramètre:**
>>

> >> - **credentialId:** L'id généré lors de votre appel api post with csr ("fromCSR": true,)

Reviewer: ```suggestion
>> - `credentialId` : ID généré lors de votre appel API POST with CSR ("fromCSR": true,)
```

> >> - **okmsId:** L'id de votre KMS OVHcloud.

Reviewer: ```suggestion
>> - `okmsId` : ID de votre KMS OVHcloud.
```

>> 

> >> Puis lancer la commande Awk ci-dessous pour le formater afin que vSphere puisse le lire :

Reviewer: ```suggestion
>> Lancez ensuite la commande awk ci-dessous pour le formater afin que vSphere puisse le lire :
```

>>
>> ```Shell
>>  awk '{gsub(/\\n/,"\n")}1' csr_signé.pem
>> ```
>>

> >> Copier le retour de la commande et coller là dans vSphere on OVHcloud.

Reviewer: ```suggestion
>> Copiez le retour de la commande et collez-le dans vSphere on OVHcloud.
```

>> 

> >> ![Trust KMS server with CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_upload_signed_csr_2-optim-resize.png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with CSR](images/okms_vsphere_upload_signed_csr_2-optim-resize.png){.thumbnail}
```

>>

> >> Vous pouvez verifier que la connection à bien était établie en sélectionnant votre fournisseur de clé KMS OVHcloud.

Reviewer: ```suggestion
>> Vérifiez que la connection à bien été établie en sélectionnant votre fournisseur de clé KMS OVHcloud.
```

>> 
>> L'option `Connecté`{.action} doit être cochée.
>>

> >> ![Trust KMS server with CSR](/pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_okms_vm-encrypt/images/okms_vsphere_okms_validation.png){.thumbnail}

Reviewer: ```suggestion
>> ![Trust KMS server with CSR](images/okms_vsphere_okms_validation.png){.thumbnail}
```

>>

> ## Étape 5 - Création d'une politique IAM

Reviewer: ```suggestion
### Étape 5 - Création d'une politique IAM
```
